pkg/generate/code/late_initialize.go

```diff
@@ -0,0 +1,188 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package code
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
+	"github.com/aws-controllers-k8s/code-generator/pkg/model"
+)
+
+// FindLateInitializedFieldsWithDelay outputs the code to create a map of fieldName to
+// late intialization delay in seconds.
+func FindLateInitializedFieldsWithDelay(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	//Sample output
+	//var lateInitializeFieldToDelaySeconds = map[string]int{"Name": 0}
+	out := ""
+	indent := strings.Repeat("\t", indentLevel)
+	fieldNameToConfig := cfg.ResourceFields(r.Names.Original)
+	if len(fieldNameToConfig) > 0 {
+		fieldNameToDelaySeconds := make(map[string]int)
+		sortedFieldNames := make([]string, 0)
+		for fName, fConfig := range fieldNameToConfig {
+			if fConfig != nil && fConfig.LateInitialize != nil {
+				fieldNameToDelaySeconds[fName] = fConfig.LateInitialize.DelaySeconds
+				sortedFieldNames = append(sortedFieldNames, fName)
+			}
+		}
+		sort.Strings(sortedFieldNames)
+		lateInitFieldToDelayValues := ""
+		if len(sortedFieldNames) > 0 {
+			for _, fName := range sortedFieldNames {
+				lateInitFieldToDelayValues += fmt.Sprintf("\"%s\":%d,",fName, fieldNameToDelaySeconds[fName])
+			}
+			out += fmt.Sprintf("%svar lateInitializeFieldToDelaySeconds = map[string]int{%s}\n", indent, lateInitFieldToDelayValues)
+		}
+	}
+	return out
+}
+
+// lateInitializedFieldNames returns the field names which have LateInitialization configuration inside generator config
+func lateInitializedFieldNames(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+) []string {
+	fieldNames := make([]string, 0)
+	fieldNameToConfig := cfg.ResourceFields(r.Names.Original)
+	if len(fieldNameToConfig) > 0 {
+		for fName, fConfig := range fieldNameToConfig {
+			if fConfig != nil && fConfig.LateInitialize != nil {
+				fieldNames = append(fieldNames, fName)
+			}
+		}
+	}
+	return fieldNames
+}
+
+// LateInitializeFromReadOne returns the gocode to set LateInitialization fields from the ReadOne output
+// Field path separated by '.' indicates members in a struct
+// Field path separated by '..' indicates member/key in a map
+// Note: Unlike Map, updating individual element of a list is not supported. LateInitializing complete list is supported.
+func LateInitializeFromReadOne (
```

Comment:
Thinking we can use this function https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/crd.go#L444-L477 or something similar to verify that we are not passing a wrong field path or a to slice/non-struct type (should probably panic)

Reply:
But we do support Map here, so that function will need some modification. I agree to adding more static validation, adding a TODO here. which I will address along with retry-backoff task. 

Also currently the generate code will fail to compile for service team if their path is not correct, so that's one safety mechanism currently.